src/V3Const.cpp

```diff
@@ -3302,6 +3302,12 @@ class ConstVisitor final : public VNVisitor {
                 } else {
                     nodep->unlinkFrBack();
                 }
+
+                if (!m_modp || !m_modp->hasParameterList()) {
+                    nodep->v3warn(UNUSEDLOOP, "Loop is not used and will be optimized out");
```

Comment:
It would be more informative to explain that "condition of loop is always false" (instead of the loop body having no effect).

Reply:
Agreed. I'll include it in this message, but I also want to emphasize that this loop will be removed from simulation flow.